require.requireActual no longer seems to use moduleNameMapper when resolving path

<!-- Love Jest? Please consider supporting our collective: 👉  https://opencollective.com/jest/donate -->

## 💥 Regression Report

`require.requireActual` no longer seems to use `moduleNameMapper`  when resolving path. Seems to be caused by this change https://github.com/facebook/jest/pull/7404/files#diff-5b4dfdf819aec7c512e60abc8ed754b5R850

## Last working version

Worked up to version: 23.6.0

Stopped working in version: 24.0.0

## To Reproduce

Add any `moduleNameMapper` config, EG: 
```json
{
  "moduleNameMapper": {
    "/lib$": "<rootDir>/mapped-lib.js"
  }
}
```

Create mapped module at destination path

```js
// mapped-lib.js
module.exports = {}
```

`require.requireActual` the source path

```js
// add.test.js
require.requireActual('./lib')
```

## Expected behavior

Mapped module should be loaded without error

## Link to repl or repo (highly encouraged)

Working repl in Jest 22.1.2
https://repl.it/repls/UnusedBoldNlp

In Jest 24.1.0 this errors:

```
● Test suite failed to run

    Cannot find module './lib' from 'add.test.js'
```

## Run `npx envinfo --preset jest`

Paste the results here:

```bash
System:
    OS: macOS 10.14
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
  Binaries:
    Node: 10.13.0 - ~/.nvm/versions/node/v10.13.0/bin/node
    Yarn: 1.12.3 - ~/.nvm/versions/node/v10.13.0/bin/yarn
    npm: 6.4.1 - ~/.nvm/versions/node/v10.13.0/bin/npm
  npmPackages:
    jest: 24 => 24.1.0
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -27,6 +27,7 @@
 - `[jest-resolve]`: Remove internal peer dependencies ([#8215](https://github.com/facebook/jest/pull/8215))
 - `[jest-snapshot]`: Remove internal peer dependencies ([#8215](https://github.com/facebook/jest/pull/8215))
 - `[jest-resolve]` Fix requireActual with moduleNameMapper ([#7981](https://github.com/facebook/jest/issues/7981))
+- `[jest-resolve]` Fix requireActual with moduleNameMapper ([#7981](https://github.com/facebook/jest/pull/8210))
 
 ### Chore & Maintenance
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -28,6 +28,7 @@
 - `[jest-snapshot]`: Remove internal peer dependencies ([#8215](https://github.com/facebook/jest/pull/8215))
 - `[jest-resolve]` Fix requireActual with moduleNameMapper ([#7981](https://github.com/facebook/jest/issues/7981))
 - `[jest-resolve]` Fix requireActual with moduleNameMapper ([#7981](https://github.com/facebook/jest/pull/8210))
+- `[jest-resolve]` Fix requireActual with moduleNameMapper ([#8210](https://github.com/facebook/jest/pull/8210))
 
 ### Chore & Maintenance
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -26,6 +26,7 @@
 - `[jest-resolve-dependencies]`: Remove internal peer dependencies ([#8215](https://github.com/facebook/jest/pull/8215))
 - `[jest-resolve]`: Remove internal peer dependencies ([#8215](https://github.com/facebook/jest/pull/8215))
 - `[jest-snapshot]`: Remove internal peer dependencies ([#8215](https://github.com/facebook/jest/pull/8215))
+- `[jest-resolve]` Fix requireActual with moduleNameMapper ([#7981](https://github.com/facebook/jest/issues/7981))
 
 ### Chore & Maintenance
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -26,8 +26,6 @@
 - `[jest-resolve-dependencies]`: Remove internal peer dependencies ([#8215](https://github.com/facebook/jest/pull/8215))
 - `[jest-resolve]`: Remove internal peer dependencies ([#8215](https://github.com/facebook/jest/pull/8215))
 - `[jest-snapshot]`: Remove internal peer dependencies ([#8215](https://github.com/facebook/jest/pull/8215))
-- `[jest-resolve]` Fix requireActual with moduleNameMapper ([#7981](https://github.com/facebook/jest/issues/7981))
-- `[jest-resolve]` Fix requireActual with moduleNameMapper ([#7981](https://github.com/facebook/jest/pull/8210))
 - `[jest-resolve]` Fix requireActual with moduleNameMapper ([#8210](https://github.com/facebook/jest/pull/8210))
 
 ### Chore & Maintenance

--- a/packages/jest-resolve/src/index.ts
+++ b/packages/jest-resolve/src/index.ts
@@ -195,11 +195,9 @@ class Resolver {
     options?: Resolver.ResolveModuleConfig,
   ): Config.Path {
     const dirname = path.dirname(from);
-    const module = this.resolveModuleFromDirIfExists(
-      dirname,
-      moduleName,
-      options,
-    );
+    const module =
+      this.resolveStubModuleName(from, moduleName) ||
+      this.resolveModuleFromDirIfExists(dirname, moduleName, options);
     if (module) return module;
 
     // 5. Throw an error if the module could not be found. `resolve.sync` only

--- a/packages/jest-runtime/src/__tests__/runtime_require_actual.test.js
+++ b/packages/jest-runtime/src/__tests__/runtime_require_actual.test.js
@@ -23,4 +23,17 @@ describe('Runtime requireActual', () => {
       );
       expect(exports.isManualMockModule).toBe(false);
     }));
+
+  test('requireActual with moduleNameMapper', () =>
+    createRuntime(__filename, {
+      moduleNameMapper: {
+        '^testMapped/(.*)': '<rootDir>/mapped_dir/$1',
+      },
+    }).then(runtime => {
+      const exports = runtime.requireActual(
+        runtime.__mockRootPath,
+        'testMapped/moduleInMapped',
+      );
+      expect(exports).toBe('in_mapped');
+    }));
 });

